js/background.js

```diff
@@ -1847,13 +1823,12 @@
           ConversationController.getOrCreateAndWait(d, 'private')
         )
       );
+      // triger session request with every devices of that user
+      // when we do not have a session with it already
       deviceConversations.forEach(device => {
-        if (device.isFriendRequestStatusNoneOrExpired()) {
-          libloki.api.sendAutoFriendRequestMessage(device.id);
-        } else {
-          // Accept any pending friend requests if there are any
-          device.onAcceptFriendRequest({ blockSync: true });
-        }
+        libsession.Protocols.SessionProtocol.sendSessionRequestIfNeeded(
+          new libsession.Types.PubKey(device.id)
+        );
```

Comment:
I think if a session request is already in progress the process should fix itself (using the latest timestamp), right?

Reply:
Well if a session request is already in progress, `hasSentSessionRequest` will return `true` and thus it'll just return early.